src/ex_getln.c

```diff
@@ -4362,7 +4355,42 @@ f_getcmdcomplpat(typval_T *argvars UNUSED, typval_T *rettv)
 f_getcmdcompltype(typval_T *argvars UNUSED, typval_T *rettv)
 {
     rettv->v_type = VAR_STRING;
-    rettv->vval.v_string = get_cmdline_completion();
+
+    if (argvars[0].v_type != VAR_UNKNOWN)
+    {
+	 char_u		*pat;
+	 expand_T	xpc;
+	 int		cmdline_len;
+
+	if (in_vim9script() && check_for_string_arg(argvars, 0) == FAIL)
+	    return;
+
+	pat = tv_get_string(&argvars[0]);
+	if (check_for_string_arg(argvars, 0) == FAIL)
+	    return;
```

Comment:
It's more common to add a `check_for_opt_string_arg(argvars, 0)` at the top of the function instead. And there is no need for the `in_vim9script()` condition, since you are calling `check_for_string_arg(argvars, 0)` again without it anyway.